### PR TITLE
156724662 Authority pre notice view

### PR DESCRIPTION
**Pull request by tatut**
# Added
- Pre notice listing and detail view for transit authority users

### CODE REVIEW

File: database/src/main/resources/db/migration/V1_77__pre_notice_comments.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE pre_notice_comment (
+  id SERIAL PRIMARY KEY,
+  "pre-notice-id" INTEGER REFERENCES pre_notice (id),
+  "created-by" TEXT REFERENCES "user" (id),
+  created TIMESTAMP WITH TIME ZONE DEFAULT NOW(),
+  "comment" TEXT
```
Comment (rckopteri) [inline database/src/main/resources/db/migration/V1_77__pre_notice_comments.sql]: No need to use quotation marks

Comment (tatut) [reply database/src/main/resources/db/migration/V1_77__pre_notice_comments.sql]: I thought that it was a reserved word, but it's not. Good catch.

File: ote/src/cljc/ote/db/user.cljc [hunk 2]
```diff
@@ -0,0 +1,20 @@
+(ns ote.db.user
+  "Datamodel for user table"
+  (:require [clojure.spec.alpha :as s]
+            #?(:clj [ote.tietokanta.specql-db :refer [define-tables]])
```
Comment (rckopteri) [inline ote/src/cljc/ote/db/user.cljc]: Pikkuisen ottaa silmään vielä tuo viimeinen suomenkielinen jäänne.

Comment (tatut) [reply ote/src/cljc/ote/db/user.cljc]: Otetaanpa härkää sarvista, vaihdanpa sen nimen samalla.

File: ote/src/cljs/ote/app/controller/pre_notices.cljs [hunk 3]
```diff
@@ -13,21 +13,25 @@
...
 ;; Load the pre-notices that are available
 (tuck/define-event LoadOrganizationPreNotices []
   {:path [:pre-notices]}
   (comm/get! "pre-notices/list"
              {:on-success (tuck/send-async! ->LoadPreNoticesResponse)
-              :on-failure (tuck/send-async! ->LoadPreNoticesFailure)})
+              :on-failure (tuck/send-async! ->ServerError)})
   :loading)
 
 (tuck/define-event LoadAuthorityPreNotices []
```
Comment (rckopteri) [inline ote/src/cljs/ote/app/controller/pre_notices.cljs]: I would suggest that in this point we should move all elykeskus related stuff away from this controller and create a new one. So we can keep our controllers short and clean.

File: ote/src/cljs/ote/ui/common.cljs [hunk 4]
```diff
@@ -166,3 +169,14 @@
...
+
+(defn gravatar
+  ([email]
+   (gravatar {:size 32 :default "mm"} email))
+  ([{:keys [size default]} email]
+   (let [hash (-> (goog.crypt.Md5.)
+                  (doto (.update email))
+                  .digest
+                  goog.crypt/byteArrayToHex)]
+     [:img {:src (str "https://www.gravatar.com/avatar/" hash "?s=" size "&d=" default)}])))
```
Comment (rckopteri) [inline ote/src/cljs/ote/ui/common.cljs]: Nice!

File: ote/src/cljs/ote/views/pre_notices/authority_listing.cljs [hunk 5]
```diff
@@ -10,9 +10,73 @@
...
+                    [:pre-notice-list-page :headers])]
+    [ui/dialog
+     {:open true
+      :modal true
+      :auto-scroll-body-content true
+      :title   (tr [:pre-notice-list-page :pre-notice-dialog :label])
+      :actions [(r/as-element
+                 [ui/flat-button
+                  {:label     (tr [:buttons :close])
+                   :secondary true
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/pre_notices/authority_listing.cljs]: can button be both? primary and secondary?

**Pull request merged**